routes/api/v1/favorites.js

```diff
@@ -10,38 +10,57 @@ const { URL, URLSearchParams } = require('url');
 
 const Geocode = require('../../../lib/objects/geocode');
 
-router.post('/', function(req, res, next) {
-  let location = req.body.location;
-  let apiKey = req.body.api_key;
+router.post('/', async function(req, res, next) {
+  let location = req.body.location || '';
+  let apiKey = req.body.api_key || '';
```

Comment:
I like that these are memoized, that's a great catch for improperly formatted request bodies.